Mathlib/Tactic/Linter/TextBased.lean

```diff
@@ -0,0 +1,203 @@
+
+/-- The error code for a given style error. Keep this in sync with `parse?_errorContext` below! -/
+-- FUTURE: we're matching the old codes in `lint-style.py` for compatibility;
+-- in principle, we could also print something more readable.
+def StyleError.errorCode (err : StyleError) : String := match err with
+  | StyleError.fileTooLong _ _ => "ERR_NUM_LIN"
+
+/-- Context for a style error: the actual error, the line number in the file we're reading
+and the path to the file. -/
+structure ErrorContext where
+  /-- The underlying `StyleError`-/
+  error : StyleError
+  /-- The line number of the error -/
+  lineNumber : ℕ
+  /-- The path to the file which was linted -/
+  path : System.FilePath
+
+/-- The parts of a `StyleError` which are considered when matching against the existing
+  style exceptions: for example, we ignore the particular line length of a "line too long" error. -/
+def StyleError.normalise (err : StyleError) : StyleError := match err with
+  -- NB: keep this in sync with `parse?_errorContext` below.
+  | StyleError.fileTooLong _ _ => StyleError.fileTooLong 0 0
+
+/-- Careful: we do not want to compare `ErrorContexts` exactly; we ignore some details. -/
+instance : BEq ErrorContext where
+  beq ctx ctx' :=
+      -- XXX: `lint-style.py` was calling `resolve()` on the path before before comparing them
+      -- should we also do so?
+      ctx.path == ctx'.path
+      -- We completely ignore line numbers of errors. Not sure if this is best.
+      -- We normalise errors before comparing them.
+      && (ctx.error).normalise == (ctx'.error).normalise
+
+/-- Output the formatted error message, containing its context. -/
+def outputMessage (errctx : ErrorContext) : String :=
+  -- We are outputting for github: duplicate file path, line number and error code,
+  -- so that they are also visible in the plain text output.
+  let path := errctx.path
+  let nr := errctx.lineNumber
+  let code := errctx.error.errorCode
+  s!"::ERR file={path},line={nr},code={code}::{path}:{nr} {code}: {errctx.error.errorMessage}"
+
+/-- Try parsing an `ErrorContext` from a string: return `some` if successful, `none` otherwise. -/
+def parse?_errorContext (line : String) : Option ErrorContext := Id.run do
+  -- Lines starting with "-- " are treated as a comment and ignored.
+  if line.startsWith "-- " then
+    return none
+  let parts := line.split (fun c ↦ c == ' ')
+  match parts with
+    | filename :: ":" :: "line" :: _line_number :: ":" :: error_code :: ":" :: error_message =>
+      -- Turn the filename into a path. XXX: is there a nicer way to do this?
+      -- Invariant: `style-exceptions.txt` always contains Unix paths
+      -- (because, for example, in practice it is updated by CI, which runs on unix).
+      -- Hence, splitting and joining on "/" is actually somewhat safe.
+      let path : System.FilePath := System.mkFilePath (filename.split (fun c ↦ c == '/'))
```

Comment:
Eyes on this are appreciated: the scenario to avoid is
- Windows user updates the style-exceptions file, to contain system paths using back-slashes
- this code tries to read the file, but doesn't split: this is still fine on its own (as the back-slashes are fine on Windows)
- the updated code gets, pushed; somebody on Mac or Linux tries to run the updated file and fails
In other words: can I split on all separators instead?

Reply:
I rewrote this check: the new version should be much safer now.